Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc

```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
-    myTree_->Branch("mysubd",   &mysubd,   "mysubd/I");
-    myTree_->Branch("depth",    &depth,    "depth/I");
-    myTree_->Branch("ieta",     &ieta,     "ieta/I");
-    myTree_->Branch("iphi",     &iphi,     "iphi/I");
-    myTree_->Branch("mom0_MB",  &mom0_MB,  "mom0_MB/F");
-    myTree_->Branch("mom1_MB",  &mom1_MB,  "mom1_MB/F");
-    myTree_->Branch("mom2_MB",  &mom2_MB,  "mom2_MB/F");
-    myTree_->Branch("mom3_MB",  &mom3_MB,  "mom3_MB/F");
-    myTree_->Branch("mom4_MB",  &mom4_MB,  "mom4_MB/F");
-    myTree_->Branch("trigbit",  &trigbit,  "trigbit/I");
-    myTree_->Branch("rnnumber", &rnnumber, "rnnumber/D");
-  }
-  myTree1_      = fs_->make<TTree>("RecJet1","RecJet1 Tree");
-  myTree1_->Branch("rnnum_",   &rnnum_,    "rnnum_/D");
-  myTree1_->Branch("HBHEsize", &HBHEsize,  "HBHEsize/I");
-  myTree1_->Branch("HFsize",   &HFsize,    "HFsize/I");
-
-  myMap_.clear();
+  std::vector<int> iarray;
+  edm::ParameterSetDescription desc;
+  desc.add<bool>("RunNZS",    true);
+  desc.add<bool>("Noise",     false);
+  desc.add<double>("ELowHB",  4);
+  desc.add<double>("EHighHB", 100);
+  desc.add<double>("ELowHE",  4);
+  desc.add<double>("EHighHE", 150);
+  desc.add<double>("ELowHF",  10);
+  desc.add<double>("EHighHF", 150);
+  desc.addUntracked<double>("EMin",2.0);
+  desc.addUntracked<double>("RunMin",303441.5);
+  desc.addUntracked<double>("RunMax",304825.5);
+  desc.addUntracked<std::vector<int> >("TriggerBits", iarray);
+  desc.addUntracked<bool>("IgnoreL1",        false);
+  desc.addUntracked<std::string>("CorrFile", "CorFactor.txt");
+  desc.addUntracked<bool>("FillHisto",       false);
+  desc.addUntracked<std::vector<int> >("HcalIeta", iarray);
+  desc.addUntracked<std::vector<int> >("HcalIphi", iarray);
+  desc.addUntracked<std::vector<int> >("HcalDepth", iarray);
+  desc.add<edm::InputTag>("hbheInputMB", edm::InputTag("hbherecoMB"));
+  desc.add<edm::InputTag>("hfInputMB",   edm::InputTag("hfrecoMB"));
+  descriptions.add("recAnalyzerMinbias",desc);
 }
 
-//  EndJob
-//
-void RecAnalyzerMinbias::endJob() {
-
-  if (!fillHist_) {
-    cells = 0;
-    for (std::map<std::pair<int,HcalDetId>,myInfo>::const_iterator itr=myMap_.begin(); itr != myMap_.end(); ++itr) {
-      edm::LogInfo("AnalyzerMB") << "Fired trigger bit number "<<itr->first.first;
-#ifdef DebugLog
-      std::cout << "Fired trigger bit number "<<itr->first.first << std::endl;
-#endif
-      myInfo info = itr->second;
-      if (info.theMB0 > 0) { 
-	mom0_MB  = info.theMB0;
+ void RecAnalyzerMinbias::beginJob() {
+
+   std::string hc[5] = {"Empty", "HB", "HE", "HO", "HF"};
+   char        name[700], title[700];
+   hbhe_ = fs_->make<TH2D>("hbhe","Noise in HB/HE",61,-30.5,30.5,72,0.5,72.5);
+   hb_   = fs_->make<TH2D>("hb",  "Noise in HB",61,-16.5,16.5,72,0.5,72.5);
+   he_   = fs_->make<TH2D>("he",  "Noise in HE",61,-30.5,30.5,72,0.5,72.5);
+   hf_   = fs_->make<TH2D>("hf",  "Noise in HF",82,-41.5,41.5,72,0.5,72.5);
+   int nbin = (int)(runMax_-runMin_+0.2);
```

Comment:
No C-style cast, please.
This also raises again the question of the usefulness of doubles as run numbers.